extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/policy/PolicyApiService.java

```diff
@@ -0,0 +1,63 @@
+/*
+ *  Copyright (c) 2022 sovity GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       sovity GmbH - initial API and implementation
+ *
+ */
+
+package de.sovity.edc.ext.wrapper.api.ui.pages.policy;
+
+
+import de.sovity.edc.ext.wrapper.api.common.mappers.PolicyMapper;
+import de.sovity.edc.ext.wrapper.api.common.model.UiPolicyDto;
+import jakarta.validation.constraints.NotNull;
+import lombok.RequiredArgsConstructor;
+import org.eclipse.edc.api.model.IdResponse;
+import org.eclipse.edc.connector.policy.spi.PolicyDefinition;
+import org.eclipse.edc.connector.spi.policydefinition.PolicyDefinitionService;
+import org.eclipse.edc.policy.model.Policy;
+import org.eclipse.edc.spi.query.QuerySpec;
+
+import java.util.List;
+
+
+@RequiredArgsConstructor
+public class PolicyApiService {
+
+
+    private final PolicyDefinitionService policyDefinitionService;
+    private final PolicyMapper policyMapper;
+
+    public List<UiPolicyDto> getPolicies() {
+        var policies = getAllPolicies();
+
+        return policies.stream().map(policy -> {
+            var entry = policyMapper.buildPolicyDto(policy);
+            return entry;
+        }).toList();
+    }
+
+    private List<Policy> getAllPolicies() {
```

Comment:
With all the business/product/page naming vs core edc entity naming it is understandable to get confused.

The "policy page" is for managing "policy definitions", meaning, a Policy + Policy Definition ID.

You might have to rework some objects and stuff, as creating a "policy" means creating a "policy definition", which will also require a policy definition ID.

Also the resulting objects will be PolicyDefinitions.

Whether you'll decide to rename your API objects to PolicyDefinitions or not is up to you.

E.g. when fetching another connector's catalog, you get data offers, which contain a policy. At that point you won't have the other connector's internal policy definition's ID. However, when you create a contract definition, the only way you can supply your policy is by specifying a policy definition's ID.

Meaning:
- Policy = Policy JSON-LD, ephermal
- Policy Definition = Persisted Policy + Chosen Policy Definition ID
- Creating a Contract Definition requires a persisted Policy Definition
- Contract Negotiations contain Policies (embedded JSON only, no policy definition ID, as the data offer might come from another connector)
- Contract Offers contain Policies (embedded JSON only, no policy definition ID, the data offers definitely come from other connectors)